### ISSUE TITLE
monitor showing green status despite no data in ES

### ISSUE THREAD

**Issue by adamstauffer** [dunbarcyber/cyphon#182]
I have two monitors created, one for the Twitter data, which is set up per the tutorial, and one for my Nessus data which I am ingesting through logstash, sending to ES and both RabbitMQ queues for monitors and watchdogs. It looks like Cyphon crashed this weekend due to some errors related to reaching the DB connection limit (maybe I'll make another ticket about this if I can track down closer to root cause).

In this state, there's no data in ES for both of my configured distilleries in the past 2 days (elasticsearch.cyphon.social and elasticsearch.nessus.nessus). Both of my monitors are configured to alert if no data is detected in 24 hours, however they show as status = Green with the following information:

```
monitor_twitter
Document id	AV3PTlnOMCfsSfeAWNgx
Last healthy	Aug. 10, 2017, 11:20 p.m.
Last updated	Aug. 14, 2017, 11:52 a.m.
```

Sure enough, the document with that ID in Elasticsearch shows the following relevant data:

```
_id |   | AV3PTlnOMCfsSfeAWNgx
created_date |   | August 10th 2017, 22:20:03.000
_saved_date |   | 2017-08-11T03:20:04.799811+00:00
```

Similarly, my nessus_monitor shows the following:

```
monitor_nessus
Document id	87eb75a6-6d37-4220-ad61-c772b12154ba
Last healthy	Aug. 11, 2017, 12:10 p.m.
Last updated	Aug. 14, 2017, 12:03 p.m.
```

Elasticsearch document:
```
_id |   | 87eb75a6-6d37-4220-ad61-c772b12154ba
@timestamp |   | August 11th 2017, 11:10:26.700
```

**Comment by lhadjchikh** [comment]
I'm not sure why the monitors would be green, given the values for ``last_healthy`` and ``last_updated``. If they're supposed to be set to 24 hours, can you double check that the units are indeed set to "hours"? If it looks correct, try switching to a shorter interval (say, 1 minute), and see if the monitors switch to red within a couple minutes. I'm not sure what's the issue is yet, so this would help to troubleshoot it.

I also don't know why the nessus index in Elasticsearch would be missing data. If you're saving the data through Logstash, it shouldn't be affected by Cyphon going down (this would only affect Alert generation). The social index not having data makes sense, since that data is saved by Cyphon, but Cyphon doesn't have anything to do with saving the nessus data. Did any of your other containers generate errors messages (logstash, elasticsearch, rabbit, etc.)?

**Comment by adamstauffer** [comment]
I tried changing the monitor's time unit to 'seconds' and it successfully changed the status to Red for both nessus and twitter. I have been playing with the time units and intervals to try to get it to go back to green (ex. 2147483647 hours) and the status still won't change. I stopped the Nessus scanning for now so I can try to isolate the issue without having too many variables

**Comment by lhadjchikh** [comment]
Thanks for the info; I'll look into this further and let you know what I find!

**Comment by lhadjchikh** [comment]
The bug that prevents monitors from going back to green after increasing the interval should be fixed by this [patch](https://github.com/dunbarcyber/cyphon/releases/tag/1.4.1). I'm still unsure of the cause of your original issue, though, in which the monitors fail to go red. We'll try to reproduce it if we can.

**Comment by adamstauffer** [comment]
update - I pulled master and I'm at commit 345507c590411d5da0bcaa52615b46a3535887fe

I re-created my Nessus monitor (with 1 day) and ingested some data and the status changed from None to Green. I just changed the time on the monitor to 8 minutes (and there's no data since 8 minutes) yet the status is not changing:

```
Last updated	Aug. 15, 2017, 8:29 p.m.
Last healthy	Aug. 15, 2017, 6:28 p.m.
```

 so I think this bug still exists. I may try to debug some of the code tomorrow but if you have any ideas I can try those first

**Comment by lhadjchikh** [comment]
I haven't been able to reproduce this, unfortunately (testing with the latest Docker image, based on 1.4.1). Are you using a Docker deployment in a development environment (building the image locally from the Git repository)? If so, I think Django should have automatically restarted the server when it detected code changes, but you might try rebuilding the Docker image, just in case.

I did make an additional change in the dev branch that will allow you to see changes to a monitor's status as soon as changes to the monitor are saved (rather than waiting for the ``run_health_check`` Celery task to run). See https://github.com/dunbarcyber/cyphon/pull/186/. This should make it easier to test changes to monitor intervals in the future.

**Comment by adamstauffer** [comment]
I switched to dev branch, modified the monitor (to 20 minutes from 8 minutes) and it changed to Red. When I switch it to 8 days (which I have data in that timeframe) it goes to Green, so I think this is working better in the dev branch. I also tested ensuring that the monitor will change from Green to Red and back again automatically (without modifying the monitor time interval). I think this is good to close
